src/draw/sw/lv_draw_sw_gradient.h

```diff
@@ -58,6 +64,140 @@ lv_grad_t * lv_gradient_get(const lv_grad_dsc_t * gradient, int32_t w, int32_t h
  */
 void lv_gradient_cleanup(lv_grad_t * grad);
 
+/**
+ * Initialize gradient color map from a table
+ * @param grad      pointer to a gradient descriptor
+ * @param colors    color array
+ * @param fracs     position array (0..255): if NULL, then colors are distributed evenly
+ * @param opa       opacity array: if NULL, then LV_OPA_COVER is assumed
+ * @param num_stops number of gradient stops (1..LV_GRADIENT_MAX_STOPS)
+ */
+void lv_gradient_init_stops(lv_grad_dsc_t * grad, const lv_color_t colors[], const uint8_t fracs[],
+                            const lv_opa_t opa[], int num_stops);
```

Comment:
Having the color and opa next to each other seems more logical
```suggestion
void lv_gradient_init_stops(lv_grad_dsc_t * grad, const lv_color_t colors[], const lv_opa_t opa[], 
                              const uint8_t fracs[], int num_stops);
```

Reply:
My reasoning was that for complex gradient color maps specifying the stop positions may be more common than using opacity, but certainly both ways make sense.

Reply:
I see it like `array1, array_len, array2`, and `array1, array2, array_len` is cleaner to me. On other APIs we put `len` to the end too, so let's follow this here as well.

Reply:
Currently it is `array1, array2, array3, array_len`, i.e., the array length is the last argument. Your original comment was about swapping `array2` (fracs) and `array3` (opa). Now I don't understand what is wrong. Shall I swap `fracs` and `opa`?

Reply:
Oh, I'm sorry. I got confused. Originally I just meant that the color and opa should be togother.